### DISCUSSION TITLE
New blog posts not appearing

### DISCUSSION THREAD

**Discussion by ICArmenian-Mycologists** [greenelab/lab-website-template#195] [Q&A] [answered]
Hi all, 
The last time I was really able to work with this lab template with no issues was in 2020. Last year (2022) I have an issue observing new information on our website.

Recently I observed an issue with Jekyll not compiling of which I updated the two gem files on our home page (following suggestions from here https://github.com/alshedivat/al-folio/issues/1208). I've been primarily working through the website online and not on the command line (still new to this).

I'm not sure where to start to look into this issue as before it was easy adding a blog post using the proper date in the title along with some post updates and photos. 

Since 2021 I've added two posts for 2022 and have not been successful in having the posts appear on our main website (for your reference https://icarmenian-mycologists.github.io/). 
I am sure there were a lot of updates since my initial forking and I just want to make sure I have all the dependencies covered and anything else I could be missing that is delaying our website being up-to-date.

Thanks!
Tania

**Comment by vincerubinetti** [comment, marked as answer]
Hi Tania,

To verify, this is the repo you are talking about, right:
https://github.com/ICArmenian-Mycologists/ICArmenian-Mycologists.github.io

Indeed, it looks like you're using a very very old version of the template. It has changed massively since you last have used it. See the new docs site here:
https://greene-lab.gitbook.io/lab-website-template-docs

And particularly, [this section here](https://greene-lab.gitbook.io/lab-website-template-docs/advanced/update-your-template#from-pre-release) about upgrading from an old version of the template. Basically, I would strongly recommend starting a new instance from scratch, and then copying over your pages/blog posts/etc. This will be better in the long run, and would most likely prevent the issue you're running into here.

---

However, to troubleshoot your current problem.

I see you've been making changes to your repo today, but [all of the attempts to build your site in the last year or so have been failing](https://github.com/ICArmenian-Mycologists/ICArmenian-Mycologists.github.io/actions). That is, it's not just the new blog posts that aren't showing up; no new changes should be showing up.

As you saw, the error being thrown is `Github can't satisfy your Gemfile dependencies`. The pre-release (< v1.0.0) versions of the template use GitHub Pages' built-in building of Jekyll, and that building only supports certain Gems (Ruby packages) and versions of them, which are [listed here](https://pages.github.com/versions/). Looking on the internet, this same error seems to have happened to a lot of people start in March of this year, so my guess is that GitHub recently changed that list of supported packages... either removing some, or increasing the required version number (for safety reasons maybe).

It looks like you upgraded your `Gemfile` like that issue suggests, and also ran the necessary `bundle` command to update the `Gemfile.lock` file. However, in doing so it looks like you added plugins that are not on by that list of supported packages: `jekyll-spaceship`, `jekyll-sitemap`, etc. I'm guessing that you looked at the latest `lab-website-template` source code, and copied those from there? The new (> v1.0.0) version of the template can use any packages/plugins it wants because it takes a totally different approach, but you'll have to stick to plugins that are on that supported list. I believe if you remove those plugins and rerun `bundle install`, this particular error will go away.  

As for why it was failing before you updated the Gemfiles, I'm not sure.

In general, you will not be able to look at the most recent version of the template and copy over changes. It has changed too much.
